README.md

```diff
@@ -17,3 +17,329 @@ Implement a prototype job worker service that provides an API to run arbitrary L
+        drwxr-xr-x  23 root             root                        4096 Aug 17 16:01 ..
+        dr-xr-xr-x   9 root             root                           0 Aug 24 11:41 1
+        dr-xr-xr-x   9 root             root                           0 Aug 26 16:40 10
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 26 16:40 1003
+        dr-xr-xr-x   9 root             root                           0 Aug 26 17:08 1004
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 25 23:08 10202
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 25 23:09 10256
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 24 13:25 10485
+        ```
+
+* The server will only store running pids in memory and have the pass logs available in their respective \<Pid\>_\<startingTimeStamp\> folder
+
+* concurrent go routines to handle cmd executions starting and finishing 
+
+* killing of pid can be implemented can be done through sigkill via syscall
+
+* Execute Api will be a bidirection streaming api, so the server can continously send back responses, while client can send new requests
+
+* Data for query for all processes ever executed by linux worker server should be generated by iterating through file system and extracting data from
+logs and filepath using path packages
+
+* Implement simple logging for server, as in the server stdout and stderr will output into a log in the server's filesystem
+
+#### Client
+
+* client is responsible for constructing the commands to be passed to server via the Execute Api
+
+* client is responsible for interpreting response from Execute Api, and must check the response type
+
+* client is responsible of constructing a queue to generate commands to feed to the Execute Api, or can be run by run via terminal
+
+
+### Authentication
+
+* Using standard GRPC tsl/ssl encryption and authentication via certificates
+
+* we will be using grpc with mutual tls
+
+* certificates will be generated using openssl and self-signed, though in a real production environment, certificates will be generated and signed 
+by a valid certificate authority (CA). there is :
+
+    * public domains:
+
+        *  Let’s Encrypt - a free automated open certificate authority got cert generation and distribution  
+
+    * private domains:
+
+        * vault to generate signing requests, centrify for renewels of certificates
+
+### Contentious Issues
+
+* Should the client remember commands they executed and remember the pids, starting timestamp, process command?
+
+* Will logs that are too old be deleted by server, or should the logs just stay in the file system storage, should this be within the scope of the project?
+
+* For the output of logs, should the should the contents of the log be loaded into a string array in the response message, or should I leave it as string?
```

Comment:
a string or a byte array are fine

Reply:
Byte array is preferable IMO; `string` tends to imply encoded text, even if `go` doesn't generally enforce it.

Reply:
I think I'm going to try to use []byte thanks